src/main/java/org/elasticsearch/indices/recovery/RecoverySourceHandler.java

```diff
@@ -268,6 +269,9 @@ protected void doRun() {
                                     .withType(TransportRequestOptions.Type.RECOVERY)
                                     .withTimeout(recoverySettings.internalActionTimeout());
 
+                            // When we last consulted RateLimiter:
+                            long lastPauseCheckBytes = 0;
```

Comment:
Yeah, it's a judgement call. I personally would go for correctness if the complexity price is not too high. Then I don't have to worry about esoteric cases like a super high settings or many many small files. Up to you :)